connector/src/main/java/org/jboss/as/connector/subsystems/datasources/DataSourceEnable.java

```diff
@@ -78,27 +78,28 @@ public void execute(OperationContext context, ModelNode operation) {
 
         final Resource resource = context.readResourceForUpdate(PathAddress.EMPTY_ADDRESS);
         final ModelNode model = resource.getModel();
-        if (! context.isBooting())
+        if (!context.isBooting())
             model.get(ENABLED).set(true);
 
-        if (context.isNormalServer()) {
-            DataSourceStatisticsService.registerStatisticsResources(resource);
-
+        if (context.isBooting()) {
             context.addStep(new OperationStepHandler() {
                 public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
                     final ManagementResourceRegistration registration = context.getResourceRegistrationForUpdate();
                     addServices(context, operation, registration, model, isXa());
```

Comment:
WFLY-4294 says this op will fail in runtime on a server. But it looks like this one is still adding services.